outputs.tf

```diff
@@ -4,17 +4,17 @@
 
 output "vpc_name" {
   description = "Name of VPC created"
-  value       = ibm_is_vpc.vpc.name
+  value       = var.create_vpc == true ? var.name : data.ibm_is_vpc.vpc[0].name
```

Comment:
You should not use variables in outputs like this. Always pull a value from the resource itself. So change `var.name` -> `ibm_is_vpc[0].vpc.name`